lib/node_modules/@stdlib/math/stats/anova1/lib/make_anova.js

```diff
@@ -0,0 +1,162 @@
+* @param {boolean} [options.decision=false] - Indicate whether we reject the null hypothesis
+* @returns {Object} Returns the ANOVA Structure
+*/
+function anova1(x, factor) {
+	var err;
+	var means;
+	var treats;
+	var grandMean;
+	var i;
+	var sumSqTotal;
+	var sq; // Inside the double summation
+	var ssTreatment;
+	var treatment; // Index variable
+	var sumSqError;
+	var meanSumSqTreat; // Mean sum of squares
+	var meanSumSqError;
+	var fScore;
+	var numDf;
+	var denDf;
+	var pVal;
+	var out;
+	var opts;
+
+	// Initialize all vars
+	grandMean = 0;
+	sumSqTotal = 0;
+	ssTreatment = 0;
+
+	opts = copy( defaults );
+
+	// First check pre-conditions
+	if ( !isTypedArrayLike( x ) && !isNumberArray( x )) {
+		throw new TypeError( 'invalid first argument. x must be numeric array. Value: `' + x + '`.');
+	}
+
+	// Validate the options
+	if (arguments.length > 2) {
+		err = validate(opts, arguments[2]);
+		if ( err ) {
+			throw err;
+		}
+	}
+
+	if (x.length <= 1) {
+		throw new RangeError( 'length of x must be greater than or equal to 2.');
+	}
+
+	// Check if factor is an array
+	if ( !isArray( factor )) {
+		throw new TypeError(' factor argument must be an array' );
+	}
+
+	if (factor.length <= 1) {
+		throw new RangeError('Second argument must have at least 2 elements');
+	}
+
+	treats = uniqueElems(factor);
+	if (!treats) {
+		throw new RangeError('Second argument must have at least two unique elements');
+	}
+	if (treats.length <= 1) {
+		throw new RangeError('Second argument must have at least 2 unique elements');
+	}
+
+	// Check if the arrays are the same length
+	if ( x.length !== factor.length ) {
+		throw new RangeError( 'Arguments must be the same length' );
+	}
+
+	means = meanTable(x, factor);
+
+	// Get the total mean
+	for (i = 0; i < x.length; i++) {
+		grandMean += x[i];
+	}
+
+	grandMean /= x.length;
+
+	// Now get total ss
+	for (i = 0; i < x.length; i++) {
+		sq = (x[i] - grandMean) * (x[i] - grandMean);
+		sumSqTotal += sq;
+	}
+
+	/* Now get the treatment SS
+	// Groupmean - grandMean squared times number of obs
+	make an array of means so that I may loop through the indices
+	*/
+	sq = 0;
+	for (treatment in means) {
+		if (treatment in means) {
+			// Already have sq defined
+			sq = (means[treatment].mean - grandMean) *
+				(means[treatment].mean - grandMean);
+			ssTreatment += means[treatment].sampleSize * sq;
+		}
+	}
+
+	// Now for sse
+	// Sse = sst - ssTreatment
+	sumSqError = sumSqTotal - ssTreatment;
+
+	// Now get the mean sum squares and mean errer
+	meanSumSqTreat = ssTreatment / (treats.length - 1);
+
+	meanSumSqError = sumSqError / (x.length - treats.length);
+
+	// Now get the f score
+	fScore = meanSumSqTreat / meanSumSqError;
+
+	// Now get the p-value
+	numDf = treats.length - 1;
+	denDf = x.length - treats.length;
+	pVal = 1 - cdf(fScore, numDf, denDf);
+
+	// Now to make the out object
+	out = {};
+	setReadOnly(out, 'treatment_df', treats.length - 1);
```

Comment:
Can we avoid underscores in property names? We could use names like `treatSumSq` and `errMeanSq`? Not so sure about the degrees of freedom. What do you think, @frank113 and @kgryte?

Reply:
I would take inspiration from MATLAB (https://www.mathworks.com/help/stats/anova1.html), including in terms of the output data structures.

Reply:
@kgryte The MATLAB looks much cleaner than what I have; far more readable. I will update the naming to mirror it. For `Prob>F` could we simply replace it with `p`?

Reply:
Instead of `Prob>F`, let's use `pValue` to be consistent with the other stats packages.

Reply:
Yes, or `pValue`, whichever is the convention currently used my @Planeshifter elsewhere.